src/app/table/table.component.ts

```diff
@@ -153,4 +193,52 @@ export class TableComponent implements OnInit, OnChanges {
   private initSelection(): void {
     this.columns.add(ColumnSelectingConfig);
   }
+
+  private initSorting(): void {
+    if (this.config.sorting == null) {
+      return;
+    }
+
+    let columnsWithSorting: TableConfigSortingColumnInterface[] = [];
+
+    if (this.config.sorting.type === TableActions.Single) {
+      const columnWithSorting = this.config.sorting?.columns.find(
```

Comment:
To avoid two level of nesting in this case better to try like that:

```typescript
if (this.config.sorting.type === TableActions.Single && this.config.sorting?.some(col => col.order !== null).length > 0) {
   columnsWithSorting.push(columnWithSorting);
}
```

Reply:
I didn't get where you setting value `columnWithSorting`